website/public/js/services/guideServices.js

```diff
@@ -195,19 +195,23 @@ function($rootScope, User, $timeout, $state, Analytics) {
         }
       };
       step.onHide = function(){
-        var ups={};
-        if (!$rootScope.stepAwarded) $rootScope.stepAwarded = {};
-        if (!$rootScope.stepAwarded[i]) {
-          $rootScope.stepAwarded[i] = true;
-          ups['stats.gp'] = User.user.stats.gp + (step.gold || 0);
-          ups['stats.exp'] = User.user.stats.exp + (step.experience || 0);
+        var ups = {};
+        var lastKnownStep = User.user.flags.tour[k];
+
+        if (i > lastKnownStep) {
+          if (step.gold) ups['stats.gp'] = User.user.stats.gp + step.gold;
+          if (step.experience) ups['stats.exp'] = User.user.stats.exp + step.experience;
+          ups['flags.tour.'+k] = i;
         }
         if (step.final) { // -2 indicates complete
           ups['flags.tour.'+k] = -2;
-          $rootScope.stepAwarded = null;
           Analytics.track({'hitType':'event','eventCategory':'behavior','eventAction':'tutorial','eventLabel':k+'-web','eventValue':i+1,'complete':true})
         }
         User.set(ups);
+        // User.set() doesn't include a check for level changes, so manually check here.
+        if (step.experience) {
+          User.user.ops.checkStats({});
```

Comment:
I tried this route first, but was running into problems that I didn't know how to solve. Here's what happens when using the `fns` over the `ops` function:

1. Create new account
2. Step through all tutorial steps
3. Hit level 2 upon completion
4. Refresh the page
5. User is lvl1 with 151 exp

This is never the case when I use the wrapper in `ops`. Perhaps there needs to be an additional save call after using `fns.updateStats()`?